src/MSBuild.Abstractions/ProjectItemHelpers.cs

```diff
@@ -124,5 +124,12 @@ public static class ProjectItemHelpers
         public static bool IsReferencingSystemWeb(ProjectItemElement item) =>
             item.ElementName.Equals(MSBuildFacts.MSBuildReferenceName, StringComparison.OrdinalIgnoreCase)
             && item.Include.Equals(MSBuildFacts.SystemWebReferenceName, StringComparison.OrdinalIgnoreCase);
+
+        public static bool IsReferencingSettingsSingleFileGenerator(ProjectItemElement item)
+        {
+            var metadata = item.Children.FirstOrDefault(child => "Generator".Equals(child.ElementName, StringComparison.Ordinal) ) as ProjectMetadataElement;
```

Comment:
These strings should probably be pulled out into constants in `MSBuildFacts` or some similar facts class that's appropriate.